uxarray/core/grid.py

```diff
@@ -233,6 +233,16 @@ def __from_ds__(self, dataset):
         # call reader as per mesh_type
         if self.mesh_type == "exo":
             self._ds = _read_exodus(dataset, self.grid_var_names)
+
+            # Assume Exodus was read as cartesian grid and that coordinates are not set by reader, call the latlon setter
+            self._populate_lonlat_coord()
+            if len(self._ds.coords) > 2:
+                # set coordinates
+                ds = self._ds.set_coords(
+                    ["Mesh2_node_x", "Mesh2_node_y", "Mesh2_node_z"])
+            else:
+                ds = self._ds.set_coords(["Mesh2_node_x", "Mesh2_node_y"])
```

Comment:
`Mesh2_node_x`, `Mesh2_node_y` and `Mesh2_node_z` should be in degrees (lat/lon), so there should never be an instance of `Mesh2_node_z`

If you are populating the Cartesian coordinates, use `Mesh2_node_cart_x`, `Mesh2_node_cart_y`, `Mesh2_node_cart_z`

It appears that you have these flipped throughout this PR.

Reply:
I'm trying to write Cartesian mesh instead of latlon for exodus. Also referenced here: https://github.com/UXARRAY/uxarray/issues/288

Reply:
That's understood, however if you are reading/writing in Cartesian, the coordinates should be `Mesh2_node_cart_x/y/z`, since it's expected for `Mesh2_node_x/y/z` to always be in degrees.

There should also never exist a `Mesh2_node_z`, since our interpretation of the UGRID conventions defines the lat/lon coordinates are on a 2D plane. `Mesh2_node_cart_z` would be the equivalent in Cartesian.

Reply:
I don't follow, for the failing test: we're loading a ugrid file lat/lon, and writing exodus, do you mean Mesh2_node_z should not exist? My intent is to always write Cartesian coordinates for exodus output file.